### ISSUE TITLE
Non-deterministic validation error with ddx/ddy, QuadReadLaneAt and loop flow

### ISSUE THREAD

**Issue by fstrugar** [microsoft/DirectXShaderCompiler#2556]
Hello, here's a maybe weird one. Looks similar to https://github.com/Microsoft/DirectXShaderCompiler/issues/1614

Shader:
```
float4 PSMain(PSInput input) : SV_TARGET
{
    float4 color = input.color;
    [loop] for( int i = 0; i < uint(input.position.x) % 10; i++ )
        color.x += input.position.y / 10.0;
    float somethingA = ddx( color.x );
    float somethingB = QuadReadLaneAt( color.x, 0 );
    color.x += somethingA * somethingB;
    return color;
}
```
(https://github.com/fstrugar/D3D12HelloTriangle/blob/5428dfaff8eed1406fc9d68b9bb9e480d1255eec/HelloTriangle/shaders.hlsl#L35)

When using the dxcompiler.dll directly, every n-th compile (n being 0...100+) triggers the "Gradient operations are not affected by wave-sensitive data or control flow.". With dxc.exe this is rarer but happen too. This can also be observed using the Shader Playground:
http://shader-playground.timjones.io/6862ec805509ea70bd6d154829487f22
(but one has to open it many times to get the "at 0x1abfbccfa68 inside block ._crit_edge of function PSMain Gradient operations are not affected by wave-sensitive data or control flow.")

When it passes though, everything looks/works fine. If I disable validation, everything is always fine (and outputs are deterministic).

Repro (run with debugger and check debug output): https://github.com/fstrugar/D3D12HelloTriangle
It includes https://github.com/microsoft/DirectXShaderCompiler/releases/tag/v1.4.1907 binaries but the same happens with latest compiler master.

**Comment by fstrugar** [comment]
Ping - any updates on this? It seems that the validation gets triggered by almost any use of QuadReadLaneAt with more complex shaders.

**Comment by python3kgae** [comment]
This pull request will change the validation error into a warning https://github.com/microsoft/DirectXShaderCompiler/pull/2640

**Comment by fstrugar** [comment]
Hmm thanks for the update but this specific one should not even be a warning?

**Comment by python3kgae** [comment]
You're right. That will require another fix. But it will be lower priority though:(.

**Comment by pow2clk** [comment]
This was fixed by #3405
